File: core/spec/fakes/line_item.rb
require 'fakes/model'

module Spree
  class LineItem
    include FakeModel
    attr_accessor :variant, :variant_id, :price, :quantity, :adjustments

    def initialize(attributes={})
      attributes.each do |key, value|
        self.send("#{key}=", value)
      end
    end

    def adjustments
      @adjustments ||= []
    end
  end
end

File: guides/src/data/line_item.js
import * as R from 'ramda'
import VARIANT from './variant'

export default {
  id: 1,
  quantity: 2,
  price: '19.99',
  variant_id: 1,
  variant: R.merge(VARIANT, { product_id: 1 }),
  adjustments: [],
  single_display_amount: '$19.99',
  display_total: '$39.99',
  total: '39.99'
}
